Makefile.am

```diff
@@ -16,5 +16,8 @@ if ENABLE_PROFILE
     profile = profile
 endif
 
-SUBDIRS = libass $(test) $(compare) $(profile)
+if ASM
+    checkasm = checkasm
+endif
 
+SUBDIRS = libass $(test) $(compare) $(profile) $(checkasm)
```

Comment:
Not something that needs to addressed in this pr, but more as a future remark:

if we keep adding more recursive subdirs this will start to slow down dev-builds, since recursive targets are guaranteed to run in order and thus won't be parallelised. Also each subdir will be entered for all targets, regardless of whether or not it contributes to the current target.
We might want to consider switching to a non-recursive build *(via Automake's `include` we can stil use separate files for each unit, but paths will be different)*